test-result-summary-client/src/TrafficLight/TrafficLight.jsx

```diff
@@ -50,6 +54,42 @@ function TrafficLight() {
             );
         }
     }
+
+    const buildJavaVersionMap = async (data, buildType, originBuildType) => {
+        let javaVersionMap = new Map();
+        for (const childData of data) {
+            childData.buildType = buildType;
+            const buildNameWithPlatform = childData.buildName.slice(
+                0,
+                lastIndexOf('_')
+            );
+            if (!javaVersionMap.has(buildNameWithPlatform)) {
+                let grandchildrenData = await fetchData(
+                    `/api/getChildBuilds?parentId=${childData._id}`
+                );
```

Comment:
Can we call getChildBuilds parentId and buildName? buildName is from aggregateInfo.buildName. In this way, we do not need to pass in and check `originBuildType`. All builds are the expected type. 

<img width="526" height="108" alt="image" src="https://github.com/user-attachments/assets/51b55668-e5e3-47e0-84a5-78a2622093d1" />

For example: http://localhost:3001/api/getChildBuilds?parentId=689698a5afeb783b5e6f29a1&buildName=Perf_openjdk21_j9_sanity.perf_ppc64_aix_ilogodm_test